src/Tools/dotnet-monitor/Extensibility/PreDefinedExtensionRepositories.cs

```diff
@@ -0,0 +1,69 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.FileProviders;
+using Microsoft.Extensions.Logging;
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Reflection;
+
+namespace Microsoft.Diagnostics.Tools.Monitor.Extensibility
+{
+    internal static class PreDefinedExtensionRepositories
+    {
+        public static IEnumerable<Func<IServiceProvider, IExtensionRepository>> GetExtensionRepositoryDelegates(HostBuilderSettings settings)
```

Comment:
Rather than return a list of Func, add a `IServiceCollection` parameter and just add the services directly from this method. You could probably remove the `FolderExtensionRepoDefinition` notion. Maybe rename the method to `AddExtensionRepositories`.